Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -581,12 +893,80 @@ int MXC_SPI_MasterTransactionAsync(mxc_spi_req_t *req);
  * possible. The channel will be reset and returned to the system at the end of
  * the transaction.
  *
- * @param   req             Pointer to details of the transaction
+ * @param   req         Pointer to details of the transaction.
  *
  * @return  See \ref MXC_Error_Codes for the list of error return codes.
  */
 int MXC_SPI_MasterTransactionDMA(mxc_spi_req_t *req);
 
+/**
+ * @brief   Set up a non-blocking, interrupt-driven SPI controller transaction.
+ *
+ * @param   spi         Pointer to SPI instance's registers.
+ * @param   tx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   tx_fr_len   Number of frames to transmit from transmit buffer.
+ * @param   rx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   rx_fr_len   Number of frames to store in recieve buffer.
+ * @param   deassert    True(1)/False(0) whether to deassert target select at end of transactions.
+ * @param   target      Pointer to select target for SPI transaction.
+ *
+ * @return  See \ref MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_SPI_ControllerTransaction(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                                  uint8_t *rx_buffer, uint32_t rx_fr_len, uint8_t deassert,
+                                  mxc_spi_target_t *target);
+
```

Comment:
With the new naming convention, is it clear that the non-blocking functions are "async" and the AsyncHandler needs to be called when using these?

Reply:
I've added a note in the header file. Do you think the "Async" name should be added to these controller functions?